test/ng/compileSpec.js

```diff
@@ -5858,6 +5858,33 @@ describe('$compile', function() {
               expect(componentScope.owRef).toEqual({name: 'lucas', item: {name: 'martin'}});
             }));
 
+            //https://github.com/angular/angular.js/issues/15833
+            it('should work with ng-model inputs', function() {
+              var componentScope;
+
+              module(function($compileProvider) {
+                $compileProvider.directive('undi', function() {
+                  return {
+                    restrict: 'A',
+                    scope: {
+                      undi: '<'
+                    },
+                    link: function($scope) { componentScope = $scope; }
+                  };
+                });
+              });
+
+              inject(function($compile, $rootScope) {
+                element = $compile('<form name="f" undi="[f.i]"><input name="i" ng-model="a"/></form>')($rootScope);
+                $rootScope.$apply();
+                expect($rootScope.f.i).toBeDefined();
+                expect(componentScope.undi).toBeDefined();
+                //TODO: the `.not` should be removed in 1.7 due to
+                //https://github.com/angular/angular.js/commit/fd4f0111188b62773b99ab6eab38b4d2b5d8d727
+                expect(componentScope.undi).not.toBe($rootScope.f.i);
```

Comment:
If we do keep this line I'd prefer `.toBe` in 1.7 to verify that it is the exact same instance, so I figured `.not.toBe` would be better for now, if we keep this line.

Only the `$compile` line is actually necessary to reproduce the issue. Maybe I should drop everything else?